src/ppc64.h

```diff
@@ -45,18 +45,40 @@
 #define DES_COPY			0
 #define DES_BS_ASM			0
 #define DES_BS				1
-#define DES_BS_VECTOR			0
+#if defined(JOHN_ALTIVEC)
+#define DES_BS_EXPAND			3
+#if 1
+#define DES_BS_VECTOR			2
+#define DES_BS_ALGORITHM_NAME		"DES 128/128 AltiVec"
+#elif 0
+/* It is likely unreasonable to use S-box expressions requiring vsel when this
+ * operation is only available in one of the two instruction sets.
+ * So let's revert to less demanding S-box expressions. */
```

Comment:
Looks like you copied this comment from a file of mine, but did not copy the accompanying `#undef` / `#define` (would need to change the value of `DES_BS`, but in your case you already set it to 1, so this wouldn't apply). Please be more careful. Comments that no longer reflect code are at best unhelpful.

Reply:
Even more importantly, I have already tuned this AltiVec stuff on real POWER hardware. So please don't try to re-tune it on an emulator. Just leave it as-is, merging into those `.h` files only the jumbo-specific additions.